synapse/rest/admin/devices.py

```diff
@@ -84,6 +85,16 @@ async def on_DELETE(
             raise NotFoundError("Unknown user")
 
         await self.device_handler.delete_devices(target_user.to_string(), [device_id])
+
+        # TODO: don't nuke the entire cache once there is a way to associate
+        #  device_id -> introspection_token
+        # TODO: is there a way to check if the deletion request has come from MAS/OIDC
+        # authorizing server and only invalidate if that's the case?
+        if self.config.experimental.msc3861.enabled:
```

Comment:
I am assuming this is the endpoint that MAS uses to delete devices?

Reply:
Maybe in the data store?